maps.go

```diff
@@ -145,6 +154,23 @@ func MapListAppendUniqueFn[K comparable, T any](m map[K]*list.List, key K, v T,
 	}
 }
 
+// MapListCopy duplicates a map containing a list.List
+func MapListCopy[T comparable](src map[T]*list.List) map[T]*list.List {
+	fn := func(v any) (any, bool) { return v, true }
+	return MapListCopyFn(src, fn)
+}
+
+// MapListCopyFn duplicates a map containing a list.List but
+// allows the element's values to be cloned by a helper function
+func MapListCopyFn[K comparable, V any](src map[K]*list.List,
+	fn func(v V) (V, bool)) map[K]*list.List {
```

Comment:
I think this would benefit from a 
```go
if src== nil || len(src)==0 {
return make(map[K]*list.List), false
}
```

Reply:
the second argument complicates significantly the usage when "cloning" objects

Reply:
@karasz see https://github.com/darvaza-proxy/darvaza/commit/5ec539d06316f7de64446a69fe6247997a29eba0

Reply:
> @karasz see [darvaza-proxy/darvaza@5ec539d](https://github.com/darvaza-proxy/darvaza/commit/5ec539d06316f7de64446a69fe6247997a29eba0)

brain typo. Please read as
```go
return make(map[K]*list.List)
```

Reply:
@karasz `golangci-lint` didn't like the idea
```
maps.go:168:5: S1009: should omit nil check; len() for map[K]*container/list.List is defined as zero (gosimple)
        if src == nil || len(src) == 0 {
           ^
exit status 1
```

```diff
@@ -164,6 +164,11 @@ func MapListCopy[T comparable](src map[T]*list.List) map[T]*list.List {
 // allows the element's values to be cloned by a helper function
 func MapListCopyFn[K comparable, V any](src map[K]*list.List,
        fn func(v V) (V, bool)) map[K]*list.List {
+       //
+       if src == nil || len(src) == 0 {
+               return make(map[K]*list.List)
+       }
+
        out := make(map[K]*list.List, len(src))
        for k, l := range src {
                out[k] = ListCopyFn(l, fn)
```

Reply:
@karasz what about this instead?

```diff
@@ -164,9 +164,15 @@ func MapListCopy[T comparable](src map[T]*list.List) map[T]*list.List {
 // allows the element's values to be cloned by a helper function
 func MapListCopyFn[K comparable, V any](src map[K]*list.List,
        fn func(v V) (V, bool)) map[K]*list.List {
-       out := make(map[K]*list.List, len(src))
-       for k, l := range src {
-               out[k] = ListCopyFn(l, fn)
+       //
+       l := len(src)
+       if l == 0 {
+               return make(map[K]*list.List)
+       }
+
+       out := make(map[K]*list.List, l)
+       for k, lp := range src {
+               out[k] = ListCopyFn(lp, fn)
        }
        return out
 }
 ```

Reply:
Not worth it. Just drop it as it is, after more consideration feels like micro optimization.